app/crud/labels.py

```diff
@@ -0,0 +1,271 @@
+        """
+        session = session_ctx.get("session")
+        assert session, "Session must be present"
+
+        # Check that the label [group] name does not conflict with any columns already in this view
+        name = (
+            values.get("group_name") if values.get("group_name") else values.get("name")
+        )
+        name_check = (
+            f'select "{name}" from reporting.enriched_query_history where false'
+        )
+
+        try:
+            session.sql(name_check).collect()
+            assert (
+                False
+            ), "Name cannot be the same as a column in REPORTING.ENRICHED_QUERY_HISTORY. Please use a different name."
+        except snowflake.snowpark.exceptions.SnowparkSQLException as e:
+            if "invalid identifier" in e.message:
+                pass
+            else:
+                assert False, "Invalid label name."
+
+        return values
+
+    @root_validator(allow_reuse=True)
+    @classmethod
+    def validate_label_name_uniqueness(cls, values) -> "Label":
+        """
+        Validates that the labels' name (and group name) do not duplicate existing name(s) in OpsCenter.
+        :param values:
+        :return:
+        """
+        name = values.get("name", "")
+        group_name = values.get("group_name", "")
+        session = session_ctx.get("session")
+        assert session, "Session must be present"
+
+        if group_name:
+            # check if the grouped label's name conflict with :
+            #  1) another label in the same group,
+            #  2) or an ungrouped label's name.
+            #  3) another dynamic group name
+            count = session.sql(
+                """
+                SELECT COUNT(*) FROM INTERNAL.LABELS
+                WHERE
+                    (GROUP_NAME = ? AND NAME IS NULL)
+                     OR (NAME = ? AND GROUP_NAME IS NULL)
+                     OR (GROUP_NAME = ? AND NAME IS NULL)""",
+                params=(
+                    group_name,
+                    name,
+                    group_name,
+                    group_name,
+                ),
+            ).collect()[0][0]
+            assert (
+                count == 0
+            ), "Duplicate grouped label name found. Please use a distinct name."
+        else:
+            # check if the ungrouped label's name conflict with another ungrouped label, or a group with same name.
+            count = session.sql(
+                """
+                SELECT COUNT(*) FROM INTERNAL.LABELS
+                WHERE (NAME = ? AND GROUP_NAME IS NULL) OR (GROUP_NAME = ? AND GROUP_NAME IS NOT NULL)""",
+                params=(
+                    name,
+                    name,
+                ),
+            ).collect()[0][0]
+            # TODO updating a label fails after adding this assertion
+            assert count == 0, "Duplicate label name found. Please use a distinct name."
```

Comment:
This incorrectly warns on update that you are duplicating a label name. Need to fix.